internal/gouging/gouging.go

```diff
@@ -0,0 +1,124 @@
+package gouging
+
+import (
+	"fmt"
+	"math"
+
+	"gitlab.com/NebulousLabs/errors"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
+	"go.sia.tech/siad/modules"
+	"go.sia.tech/siad/types"
+)
+
+var (
+	// ErrGougingDetected is returned when price gouging was detected in one of
+	// the fields of either the price table or the host's external settings.
+	ErrGougingDetected = errors.New("price gouging detected")
+
+	downloadOverdriveEstimatePct = .1
+)
+
+type (
+	GougingChecks struct {
+		Download     GougingCheck
+		FormContract GougingCheck
+		Upload       GougingCheck
+	}
+
+	GougingCheck interface {
+		IsGouging() bool
+		error
+	}
+
+	GougingSettings struct {
+		MaxRPCPrice      types.Currency
+		MaxContractPrice types.Currency
+		MaxDownloadPrice types.Currency // per TiB
+		MaxUploadPrice   types.Currency // per TiB
+	}
+
+	check struct {
+		err error
+	}
+)
+
+func (c *check) IsGouging() bool {
```

Comment:
How about we don't expose the individual checks and have methods like `CanDownload`, `CanUpload`, `CanForm` on the `GougingChecks` which then returns a slice of errors for the corresponding gouging checks that should have passed but failed.

e..g `CanForm` would internally check `Download`, `FormContract` and `Upload`.
`CanUpload` would check the same.
`CanDownload` would only check `Download`

Reply:
I'm not sure why the `gouging` check has to make these decisions. If we keep it the way it is, the gouging checks are just exactly what they are, is the host upload gouging yes/no. The caller (autopilot/worker) can easily add a function.

```go
 func canForm(gr GougingResults) bool {
    return 
        !gr.FormContract.IsGouging() &&
        !gr.Download.IsGouging() && 
        !gr.Upload.Isgouging()
}
```

I would normally just go ahead and implement your suggestion but in this case I was initially a bit thrown by the naming.. `CanDownload` doesn't really sit well for me because I _can_ download from that host just fine, I just don't want to because I think it's too expensive.

Reply:
I already feel less strongly about now that I'm rereading my comment. There's no real good reason not to have it defined on the `GougingResults`, but I would expose the individual checks simply because I don't see a reason not to expose it.